### ISSUE TITLE
Firestick: Stremio 1.3.4 apk, cant connect to stream when using VPN

### ISSUE THREAD

**Issue by JS-World66** [Stremio/stremio-core#237]
2019 4k Amazon Firestick, 1.3.4 ARM .apk
Noticed that if I am on a VPN (I've tried a couple) and I attempt to play an item in my library, upon choosing my source it will repeatedly fail to connect, title blinking indefinitely (1hr+) as it tries.
This is strange for a few reasons. First, I can open stremio, log in, view items in my library with loaded title cover images, check season listings, etc. It is only when I attempt to view the content that it fails using a VPN connection. Second, if I disable the VPN and use my normal internet connection, Stremio seems to load streams fine, including the ones I had trouble with specifically. Third, I have verified that this is an issue localized to the firestick specifically, google play store stremio.apk on an android phone works fine with or without VPN. Desktop apps with or without VPN also works fine, and I have been using this workaround for about a few weeks now. Fourth, the flashing title of the show you're attempting to load (tried a bunch of shows with similar issues so not title specific) stays flashing for a long time. Note, user can hit the 'back' button at any time and is not stuck in the loading menu.  

VPNs I've experienced this with:
Proton VPN, Thunder VPN

Phone with Play store .apk works fine: LG G7 ThinQ, Android 9, Model: LG-G710PM

And yes, I've tried re installs/different .apk versions from the Stremio website.

**Comment by JS-World66** [comment]
Tried to install stremio 1.3.5 .apk onto firestick hoping this would patch the issue, and it is not compatible unfortunately.

**Comment by Ivshti** [comment]
> Tried to install stremio 1.3.5 .apk onto firestick hoping this would patch the issue, and it is not compatible unfortunately.

What do you mean by "not compatible"?

**Comment by JS-World66** [comment]
> 
> 
> > Tried to install stremio 1.3.5 .apk onto firestick hoping this would patch the issue, and it is not compatible unfortunately.
> 
> What do you mean by "not compatible"?

the 1.3.5 version of the stremio apk does not install on my firestick, I'm assuming the android architecture is different between phone and firestick but thats all it tells me is "this app is not compatible with your device."

**Comment by Ivshti** [comment]
@JS-World66 sounds like you downloaded the wrong architecture. Keep in mind there are two APKs on the website, one is for x64 and the other one is for ARM, you need the ARM one

**Comment by JS-World66** [comment]
No, I tried to used all of the architectures, and like you say, the ARM one was what I selected as the correct one, the rest didnt even install.

**Comment by jaruba** [comment]
@JS-World66 what OS version do you have?

**Comment by JS-World66** [comment]
My Fire TV--> About --> Software Version: Fire OS 6.2.7.7 (NS6277/3033) , Fire TV Home Version: 6242056.1

**Comment by jaruba** [comment]
@JS-World66 and where did you download Stremio v1.3.5 from? It is not available on the download page of our official website.

**Comment by JS-World66** [comment]
Stremio v1.3.5 apk was acquired from the google play store on my LG G7 Phone, I was able to get it onto the firestick from there, but to no surprise for me it didn't install on that device. I was trying everything.

**Comment by jaruba** [comment]
@JS-World66 

One thing should be clarified from the start: the only difference between Stremio v1.3.4 and v1.3.5 is a notification added for our addon competition, which is why it was never updated on our website.

You said:

> Stremio v1.3.5 apk was acquired from the google play store on my LG G7 Phone

But you also said:

> the 1.3.5 version of the stremio apk does not install on my firestick, I'm assuming the android architecture is different between phone and firestick but thats all it tells me is "this app is not compatible with your device."

Which is correct, there is too small of a chance that the Google Play APK taken from your phone will work at all on your fire stick.

But then you said:

> No, I tried to used all of the architectures, and like you say, the ARM one was what I selected as the correct one, the rest didnt even install.

Which is the confusing part for me, if you took v1.3.5 from your phone, then how did you test all architectures of this version on your Fire Stick device?

**Comment by jaruba** [comment]
I will presume that you only tried to install v1.3.5 from the G Play APK from your phone, and that there are no install issues at all with the new version, unless you have any further clarification regarding this.

Regarding your initial issue, the impossibility to stream through a VPN simply cannot be Stremio's fault, regardless of the version you are using, it should work. If it does not work, then something else is wrong, so here are some troubleshooting steps:
- You are probably trying to play something that has few peers (try playing something popular to see if it works)
- Avoid 4K releases (for the duration of your tests avoid playing 4K releases, which can also fail for other reasons too)
- Ensure that the issue you are having is not caused by a broken addon
- Uninstall all other apps except Stremio and the one for the VPN and try streaming something
- If all else fails, try factory reseting your Fire Stick device

**Comment by JS-World66** [comment]
@jaruba everything you've assessed so far is correct, the only version I was able to test multiple architectures for was 1.3.4 from the official site, I tried the 1.3.5 version from the google play store (whatever architecture that was) and tried to install on the firestick on the off chance it would work. I knew going in that it was highly unlikely. As far as dismissing my issue I will respond to your bullet point list.

- Nope, plenty of peers (>30 for every one) for all the stuff I tried to play, tried multiple sources that work fine as long as im not on the VPN.
- I have never tried to load a 4k stream from Stremio, that would take forever and I'm not interested so no worries there.
- Addons are not broken, like I said, everything I've tested on ONLY fails when VPN is connected on Firestick, same streams, same addons, works on other devices as well with or without VPN.
- Uninstall all other apps, I cant it's a firestick, theres a lot of preloaded apps that you cannot get rid of, but I'll tell you what I'll skip straight to the last bullet point suggestion
- Factory Reset for Firestick not attempted, attempting now and will return shortly.

 I dont have anything important it anyways, I literally got the Firestick pretty much explicitly for Stremio and financial charts. Be back with the results of the reset soon.

**Comment by jaruba** [comment]
A factory reset will tell us if some other app or an android setting was somehow interfering with the streaming process. If it's not that, and it's not the streams and it's not the addons, then the only thing that comes to mind is the VPN apps being the issue, but you tested 2 different VPN apps from what you said, which makes that unlikely.

From Stremio's perspective, it doesn't care if you are using a VPN or not. There are no checks for this and there are no special cases in the code for it either, it simply does the same thing in both scenarios, so it can't really be at fault in any way if one of these scenarios does not work.

This is the first time I hear of such an issue, which means that the issue itself is somehow directly related to your environment or network.

**Comment by JS-World66** [comment]
So I factory reset my firestick to no noticeable effect, and I agree that we've narrowed the field so at this point it's either the VPN, or the Operating System. Let me try some different VPNs again. It cant be the network because all my devices are connected to the same network which serves each device the exact same. At this point I'm betting it's got something the do with the OS.

**Comment by jaruba** [comment]
@JS-World66 One theory is that the OS only supports an older version of the VPN apps which have this issue. This would explain why they are working on ur phone.

**Comment by JS-World66** [comment]
Okay so sorry for the delay, but I've been testing some VPN apps on my Firestick, and havent had success with any except ProtonVPN particularly for this issue, but no matter what it's very clear the Firestick disagrees. Usually within about 10 minutes from a stream connecting and loading (if it gets there), the next time I exit to my home screen, it is empty and says "Home page currently unavailable" even though the VPN is running still and was just in use. This is a rare case where it works for a few minutes, most of the time, nothing has changed and it behaves like every other VPN app in that everything works except Stremio, Youtube, crypto charts, etc. work fine. I tried changing ProtonVPN's settings to different connection methods to test if there was a certain port or protocol that was more effective in getting Stremio to work, and using IKEv2 and OpenVPN in TCP mode were the two that seemed most successful.

I'm pleased that I found something that will work sometimes, but it is way too unstable and seemingly random to rely on. At one point I tried watching a show using the same stream with two different devices, computer and Firestick, both on a Japan VPN using OpenVPN protocol to see what would happen; the result was both the Firestick and PC were able to connect to the stream (title stopped blinking), but only the PC one would actually find peers/sources and load/play, the Firestick looked like it had a source connect and then dropped off indefinitely until network request errors occured, popping a dialogbox onto my screen. Worth noting I cant see any type of debug or networking data on the Stremio .apk, but I can see some network stuff like data rate and such on the PC one so I kinda had to "best guess" the Firestick actually connecting, my only indication being that the title stopped flashing and filled a small chunk of text as if it was loaded.

I'm open to suggestions, but at this point it seems like it's only getting more puzzling. I'd love to say it's not Stremio and that it's the OS like I think it is, but I dont have any hard evidence to back that, and it doesnt explain why Stremio is the only app I cannot stream with while connected to a VPN. And it's weird that everything in Stremio loads perfectly on the VPN until the second I pick a stream. Still baffled but enjoying Stremio on other devices just fine.
![Uploading 20210316_191959_HDR.jpg…]()

**Comment by JS-World66** [comment]
So it's been a while, got a new android tv box do see if a different device thats also Android OS responds the same way. Seems like I have more success on the tv box, rather than with the firestick using Proton VPN + Stremio, but I'm finding Proton isnt as reliable a connection as I would like either, and forever loading/ network requrest fails can still occur. A lot of other vpns straight up dont work at all with Stremio on Android but Proton is the best I've found so far as long as you connect with TCP. If eventually I find a combo of apps that work with consistency, or some other solution I'll be sure to post it.

**Comment by jaruba** [comment]
@JS-World66 

I'm happy you revived this thread, it's highly possible that I got busy and forgot to answer, and for this I'm truly sorry.

I have a crazy suggestion, can you try changing the WiFi channel on your router? I wonder if this is not somehow related to poor WiFi signal or a polluted WiFi 2.4ghz band, are there many routers in your area? It's a long shot, but Android TV devices are known too have poor WiFi connections in general. It's a long shot, but it's worth a try.
